.gitlab/kernel_version_testing/system_probe.yml

```diff
@@ -1,24 +1,58 @@
+---
 .pull_test_dockers:
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/docker_x64:$DATADOG_AGENT_BUILDIMAGES
-  needs: []
+  needs:
+    job: get_file_diff
+    artifacts: true
   tags: ["runner:docker"]
   rules:
     !reference [ .on_system_probe_changes_or_manual ]
   stage: kernel_matrix_testing
   script:
+    - set +e
+    - export SYSTEM_PROBE_PATHS=$(echo "$SYSTEM_PROBE_SRC_PATH")
+    - |+
+      for path in $SYSTEM_PROBE_PATHS
+      do
+        echo "Checking $path..."
+        echo "$DIFF_FILES" | grep "$path"
+        export exit_code=$?
+        if [ $exit_code -eq 0 ]; then
+          break
+        fi
+      done
+    - set -e
+    - if [ ${exit_code} -eq 0 ]; then
+    # Build dependencies directory
     # DockerHub login for build to limit rate limit when pulling base images
     - DOCKER_REGISTRY_LOGIN=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.$DOCKER_REGISTRY_LOGIN_SSM_KEY --with-decryption --query "Parameter.Value" --out text)
     - aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.$DOCKER_REGISTRY_PWD_SSM_KEY --with-decryption --query "Parameter.Value" --out text | docker login --username "$DOCKER_REGISTRY_LOGIN" --password-stdin "$DOCKER_REGISTRY_URL"
     # Pull base images
     - mkdir $KITCHEN_DOCKERS
     - inv -e system-probe.save-test-dockers --output-dir $KITCHEN_DOCKERS --arch $ARCH
+    - fi
   artifacts:
     expire_in: 1 day
     paths:
       - $KITCHEN_DOCKERS
   variables:
     KITCHEN_DOCKERS: $DD_AGENT_TESTING_DIR/kitchen-dockers-$ARCH
 
+get_file_diff:
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
+  tags: ["runner:main"]
+  rules:
+    !reference [ .on_system_probe_changes_or_manual ]
+  needs: []
+  stage: kernel_matrix_testing
+  script:
+    - git fetch origin main:main
+    - echo "DIFF_FILES=$(first=0;for element in $(git diff --name-only $(git merge-base $(inv release.get-release-json-value base_branch) HEAD) HEAD); do if [ $first -eq 0 ]; then first=1; else echo -n '\n'; fi; echo $element; done)" | tr -d '\n' > file_diff.env
```

Comment:
Could this be made an invoke task? I find this hard to follow.

Reply:
Done 👍